public/scripts/iobserver.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @fileoverview
+ * Lazy loading of images using an intersection observer.
+ */
+
+export default (document) => {
+  const images = document.querySelectorAll('img[d-src]');
+
+  if ('IntersectionObserver' in window) {
+    const lazyLoad = (target) => {
+      const io = new IntersectionObserver(
+        (entries, observer) => {
+          entries.forEach((entry) => {
+            if (entry.isIntersecting) {
+              const img = entry.target;
+              const src = img.getAttribute('d-src');
+              img.onload = () => {
+                img.classList.add('ld');
+              };
+              img.setAttribute('src', src);
+              img.removeAttribute('d-src');
+              observer.disconnect();
+            }
+          });
+        },
+        { threshold: [0], rootMargin: '80px' },
+      );
+
+      io.observe(target);
+    };
+    images.forEach(lazyLoad);
+  } else {
+    // Browser does not support IntersectionObserver,
+    // so we load using the loading=lazy attribute
+    for (const img of images) {
+      const src = img.getAttribute('d-src');
+      img.removeAttribute('d-src');
+      img.onload = () => {
+        img.classList.add('ld');
+      };
+      img.setAttribute('loading', 'lazy');
+      img.setAttribute('src', src);
```

Comment:
This seems unnecessary because `loading=lazy` support implies `IntersectionObserver` support.

Reply:
Thanks, removing that on #293

Reply:
This implementation is not exhaustive, if a browser like IE11 does not support `loading=lazy` just loads the images as normal ignoring the attribute. It is only added here in the case that IE11 or other browsers starts supporting it.

Reply:
> It is only added here in the case that IE11 or other browsers starts supporting it.

[IE is deprecated](https://endoflife.date/internet-explorer), while other browsers all supports it in their latest versions.

I'm just nit-picking that `'IntersectionObserver' in window` is practically always true nowadays.